### PR TITLE
Issue #15837: print diff as patch

### PR DESCRIPTION
**Pull request by mohitsatr**
Issue #15837

### CODE REVIEW

**Comment by romani** [comment]
as you fix bash violation.

please make commit that will fail checker and pitest (just remove some suppression) to let us test that copy paste  of patch is working.
as we see it works, you will remove test commit(s) and we will merge this PR.

**Comment by mohitsatr** [comment]
@romani could you tell me what this is about 
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/28574/workflows/22abffb7-ad2c-4555-855d-4853e8b8db1e/jobs/717628?invite=true#step-102-4364_37

**Comment by romani** [comment]
> Expected mode for '.sh' files is 755.

make new shell file executable. `chmod 755 your_shell.sh`

File: .github/workflows/checker-framework.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
 
       - name: Git Diff on fail
         if: failure()
-        run: git diff && git diff > target/checker-${{ matrix.profile }}.patch
+        run: ./.ci/print-diff-as-patch target/checker-${{ matrix.profile }}.patch
```
Comment (romani) [inline .github/workflows/checker-framework.yml:58]: Do we really need patch file creation?
Can we just print to output?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .github/workflows/checker-framework.yml (hunk 1 above)
Comment (mohitsatr) [reply .github/workflows/checker-framework.yml:58]: > Do we really need patch file creation?

I was thinking of the same thing. I noticed that we are archiving the .patch file for 7 days 

https://github.com/checkstyle/checkstyle/blob/f75123e43c04ea41a37cfc8e8cfe9254baf75088/.github/workflows/checker-framework.yml#L60-L66

Comment (romani) [reply .github/workflows/checker-framework.yml:58]: We can keep a file but we need to mention during failure that it is available, I already forgot about it.

I think that copy-paste command is more easy to use.

Comment (mohitsatr) [reply .github/workflows/checker-framework.yml:58]: should we mention about existence of `.patch` file in this .sh file ?

Comment (romani) [reply .github/workflows/checker-framework.yml:58]: if we keep it, yes.

Comment (mohitsatr) [reply .github/workflows/checker-framework.yml:58]: done

**Comment by romani** [comment]
patches:
https://github.com/checkstyle/checkstyle/actions/runs/11757208615/job/32754207178?pr=15863
https://github.com/checkstyle/checkstyle/actions/runs/11757208616/job/32754207250?pr=15863#step:7:9

applied to local without problems:
```
✔ ~/java/github/romani/checkstyle [mohitsatr/print-patch2 L|✔] 
$ patch -p1 <<'EOF'
> diff --git a/config/checker-framework-suppressions/checker-index-suppressions.xml b/config/checker-framework-suppressions/checker-index-suppressions.xml
> index 04726b9..16173ef 100644
> --- a/config/checker-framework-suppressions/checker-index-suppressions.xml
> +++ b/config/checker-framework-suppressions/checker-index-suppressions.xml
> @@ -2223,6 +2223,17 @@
>      </details>
>    </checkerFrameworkError>
>  
> +  <checkerFrameworkError unstable="false">
> +    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/SinglelineDetector.java</fileName>
> +    <specifier>argument</specifier>
> +    <message>incompatible argument for parameter start of Matcher.find.</message>
> +    <lineContent>while (matcher.find(startPosition)) {</lineContent>
> +    <details>
> +      found   : @GTENegativeOne int
> +      required: @NonNegative int
> +    </details>
> +  </checkerFrameworkError>
> +
>    <checkerFrameworkError unstable="false">
>      <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java</fileName>
>      <specifier>argument</specifier>
> EOF
patching file config/checker-framework-suppressions/checker-index-suppressions.xml
✔ ~/java/github/romani/checkstyle [mohitsatr/print-patch2 L|✚ 1] 
$ git diff
diff --git a/config/checker-framework-suppressions/checker-index-suppressions.xml b/config/checker-framework-suppressions/checker-index-suppressions.xml
index 04726b9..16173ef 100644
--- a/config/checker-framework-suppressions/checker-index-suppressions.xml
+++ b/config/checker-framework-suppressions/checker-index-suppressions.xml
@@ -2224,6 +2224,17 @@
   </checkerFrameworkError>
 
   <checkerFrameworkError unstable="false">
+    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/SinglelineDetector.java</fileName>
+    <specifier>argument</specifier>
+    <message>incompatible argument for parameter start of Matcher.find.</message>
+    <lineContent>while (matcher.find(startPosition)) {</lineContent>
+    <details>
+      found   : @GTENegativeOne int
+      required: @NonNegative int
+    </details>
+  </checkerFrameworkError>
+
+  <checkerFrameworkError unstable="false">
     <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java</fileName>
     <specifier>argument</specifier>
     <message>incompatible argument for parameter bitIndex of BitSet.set.</message>
✔ ~/java/github/romani/checkstyle [mohitsatr/print-patch2 L|✚ 1] 
$ 
✔ ~/java/github/romani/checkstyle [mohitsatr/print-patch2 L|✚ 1] 
 $ patch -p1 <<'EOF'
> diff --git a/config/pitest-suppressions/pitest-filters-suppressions.xml b/config/pitest-suppressions/pitest-filters-suppressions.xml
> index 0bbc17f..93116cf 100644
> --- a/config/pitest-suppressions/pitest-filters-suppressions.xml
> +++ b/config/pitest-suppressions/pitest-filters-suppressions.xml
> @@ -9,6 +9,15 @@
>      <lineContent>checkRegexp = null;</lineContent>
>    </mutation>
>  
> +  <mutation unstable="false">
> +    <sourceFile>SuppressFilterElement.java</sourceFile>
> +    <mutatedClass>com.puppycrawl.tools.checkstyle.filters.SuppressFilterElement</mutatedClass>
> +    <mutatedMethod>&lt;init&gt;</mutatedMethod>
> +    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
> +    <description>Removed assignment to member variable columnFilter</description>
> +    <lineContent>columnFilter = null;</lineContent>
> +  </mutation>
> +
>    <mutation unstable="false">
>      <sourceFile>SuppressFilterElement.java</sourceFile>
>      <mutatedClass>com.puppycrawl.tools.checkstyle.filters.SuppressFilterElement</mutatedClass>
> EOF
patching file config/pitest-suppressions/pitest-filters-suppressions.xml
✔ ~/java/github/romani/checkstyle [mohitsatr/print-patch2 L|✚ 2] 
 $ git diff
diff --git a/config/checker-framework-suppressions/checker-index-suppressions.xml b/config/checker-framework-suppressions/checker-index-suppressions.xml
index 04726b9..16173ef 100644
--- a/config/checker-framework-suppressions/checker-index-suppressions.xml
+++ b/config/checker-framework-suppressions/checker-index-suppressions.xml
@@ -2224,6 +2224,17 @@
   </checkerFrameworkError>
 
   <checkerFrameworkError unstable="false">
+    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/SinglelineDetector.java</fileName>
+    <specifier>argument</specifier>
+    <message>incompatible argument for parameter start of Matcher.find.</message>
+    <lineContent>while (matcher.find(startPosition)) {</lineContent>
+    <details>
+      found   : @GTENegativeOne int
+      required: @NonNegative int
+    </details>
+  </checkerFrameworkError>
+
+  <checkerFrameworkError unstable="false">
     <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java</fileName>
     <specifier>argument</specifier>
     <message>incompatible argument for parameter bitIndex of BitSet.set.</message>
diff --git a/config/pitest-suppressions/pitest-filters-suppressions.xml b/config/pitest-suppressions/pitest-filters-suppressions.xml
index 0bbc17f..93116cf 100644
--- a/config/pitest-suppressions/pitest-filters-suppressions.xml
+++ b/config/pitest-suppressions/pitest-filters-suppressions.xml
@@ -14,6 +14,15 @@
     <mutatedClass>com.puppycrawl.tools.checkstyle.filters.SuppressFilterElement</mutatedClass>
     <mutatedMethod>&lt;init&gt;</mutatedMethod>
     <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
+    <description>Removed assignment to member variable columnFilter</description>
+    <lineContent>columnFilter = null;</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>SuppressFilterElement.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.filters.SuppressFilterElement</mutatedClass>
+    <mutatedMethod>&lt;init&gt;</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
     <description>Removed assignment to member variable columnsCsv</description>
     <lineContent>columnsCsv = columns;</lineContent>
   </mutation>
✔
```

**Comment by romani** [comment]
patched updates are applied to PR, all commits are squashed.
no manual corrections - awesome.

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Pull request merged**